### DISCUSSION TITLE
Is there a way to make tables' column widths be automatic?

### DISCUSSION THREAD

**Discussion by postylem** [quarto-dev/quarto-cli#7399] [Q&A]
### Description

Is there a simple way to get the column widths to be automatic?  For instance, a simple table like this

```qmd
| Col1  | Col2  | Col3  |
| ----- | ----- | ----- |
| short | short | this is very long, blah blah blah, it will wrap, and its column should be wide.  it should take up most of the space.  |
```

results in a table whose columns are all ⅓ width (so, in html we get `<col style="width: 33%">` or latex we get columns with width specified each to be `{(\columnwidth - 4\tabcolsep) * \real{0.3333}}`).  

Is there a way to make columns be automatically sized, as you would get in LaTeX or HTML by default without specifying col widths?

Of course I could get what I want by something like this, but I wonder if it's possible to get this automatically by some markdown convention.

```qmd
| Col1  | Col2  | Col3  |
| ----- | ----- | -------------------------------------------------------- |
| short | short | this is very long, blah blah blah, it will wrap, and its column should be wide.  it should take up most of the space.  |
```

**Comment by cscheid** [comment]
You can specify [tbl-colwidths](https://quarto.org/docs/authoring/tables.html#column-widths) as an attribute,

```

| Col1  | Col2  | Col3  |
| ----- | ----- | ----- |
| short | short | this is very long, blah blah blah, it will wrap, and its column should be wide.  it should take up most of the space.  |

: {tbl-colwidths="[10, 10, 80]"}
```

**Reply by postylem** [reply]
This is nicer than my example, yeah, but it isn't automatic (the reason I'm wondering is, at least in HTML and LaTeX, auto widths is the _default_, so it seems plausible there'd be a way of getting that default table behavior out of pandoc/quarto, but maybe there isn't)

**Reply by cscheid** [reply]
> there'd be a way of getting that default table behavior 

No, there isn't; that's a limitation of markdown tables unfortunately.
